src/plugins/intel_npu/src/plugin/npuw/util.cpp

```diff
@@ -1405,3 +1406,105 @@ void ov::npuw::util::to_f32(const ov::Tensor& in, ov::Tensor& out) {
         break;
     }
 }
+
+void ov::npuw::util::to_f16(ov::Tensor& t) {
+    ov::Shape shape = t.get_shape();
+    NPUW_ASSERT(t.get_element_type() == ov::element::f32);
+    NPUW_ASSERT(t.get_size() % 8 == 0);
+    NPUW_ASSERT(t.is_continuous());
+
+    ov::Tensor tnew(ov::element::f16, shape);
+
+    const float* psrc = t.data<float>();
+    uint8_t* pdst = static_cast<uint8_t*>(tnew.data());
+
+    for (std::size_t i = 0; i < t.get_size() / 8; i++) {
+        __m256 vsrc = _mm256_loadu_ps(psrc);
+        __m128i vout = _mm256_cvtps_ph(vsrc, _MM_FROUND_TO_NEAREST_INT);
+        __m128i* pout = reinterpret_cast<__m128i*>(pdst);
+        _mm_storeu_si128(pout, vout);
+        psrc += 8;        // offset in sizeof(float)
```

Comment:
You load several floats, no? Why only 8?

Reply:
_mm256_loadu_ps reads 8 floats.